pkg/proc.attr.hydfab/R/proc_attr_grabber.R

```diff
@@ -636,3 +637,75 @@ grab_attrs_datasets_fsds_wrap <- function(Retr_Params,lyrs="network",overwrite=F
 
   return(ls_comids_all)
 }
+
+check_attr_selection <- function(attr_cfg_path = NULL, vars = NULL){
+  # Read in the menu of attributes available through FSDS
+  attr_menu <- yaml::read_yaml('./fsds_attr_menu.yaml')
+  
+  if(!is.null(attr_cfg_path)){
+
+    # Read in the user defined config of attributes of interest
+    # attr_cfg_path <- '../../scripts/eval_ingest/xssa/xssa_attr_config.yaml'
+    attr_cfg <- yaml::read_yaml(attr_cfg_path)
+    
+    # Determine which data sets the user specifies attributes from
+    ha_vars_sel <- base::lapply(attr_cfg$attr_select, function(x) names(x)) %>%
+      base::unlist() %>% base::grep(pattern = "ha_vars")
+    ha_vars_sel <- base::sapply(ha_vars_sel, function(x) attr_cfg$attr_select[[x]]) |> 
+      unlist() |> 
+      unname() 
+    
+    usgs_vars_sel <- base::lapply(attr_cfg$attr_select, function(x) names(x)) %>%
+      base::unlist() %>% base::grep(pattern = "usgs_vars")
+    usgs_vars_sel <- base::sapply(usgs_vars_sel, function(x) attr_cfg$attr_select[[x]]) |> 
+      unlist() |> 
+      unname() 
+    
+    # sc_vars_sel <- base::lapply(attr_cfg$attr_select, function(x) names(x)) %>%
+    #   base::unlist() %>% base::grep(pattern = "sc_vars")
+    # sc_vars_sel <- base::sapply(sc_vars_sel, function(x) attr_cfg$attr_select[[x]]) |> 
+    #   unlist() |> 
+    #   unname()
+    
+    vars_sel <- c(ha_vars_sel, usgs_vars_sel) # camels_vars_sel, sc_vars_sel
+    rm(ha_vars_sel, usgs_vars_sel) # camels_vars_sel, sc_vars_sel
+  }
+  if(is.null(attr_cfg_path) & !is.null(vars)){
+    # vars <- c("TOT_twi","TOT_PRSNOW","TOT_POPDENS90","TOT_EWT","TOT_RECHG","TOT_BFI")
+    vars_sel <- vars
+  }
+  
+  # Check if the entered variables exist in the attribute menu
+  ha_menu <- base::unlist(attr_menu$hydroatlas_attributes) |> 
+    base::names() 
+  
+  usgs_menu <- base::unlist(attr_menu$usgs_attributes) |> 
+    base::names() 
+  
+  # sc_menu <- base::unlist(attr_menu$sc_attributes) |>
+  #   base::names()
+  
+  camels_menu <- base::unlist(attr_menu$camels_attributes) |> 
+    base::names() 
+  
+  vars_menu <- c(ha_menu, camels_menu, usgs_menu) # sc_menu
+  rm(ha_menu, usgs_menu, camels_menu) # sc_menu
+  
+  # Warn the user of any requested attrs that are missing
+  missing_vars <- vars_sel[which(!vars_sel %in% vars_menu)] 
+  missing_vars_list <- paste0(missing_vars, collapse=', ')
+
```

Comment:
Also might be worth adding a boolean `verbose` argument to the function. `if verbose==TRUE`, create a `message()` describing which datasets were recognized from the user's config file.

Another message could list the variables being considered for acquisition `if verbose==TRUE`

Reply:
Done, thanks.